pkg/alerts/lcid_expiration.go

```diff
@@ -0,0 +1,243 @@
+package alerts
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// Uses CEDAR LDAP functions to get and build out recipient list for alert
+func getAlertRecipients(
+	ctx context.Context,
+	intake models.SystemIntake,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+) (*models.EmailNotificationRecipients, error) {
+
+	requesterInfo, err := fetchUserInfo(ctx, intake.EUAUserID.ValueOrZero())
+	if err != nil {
+		return nil, err
+	}
+
+	if requesterInfo == nil || requesterInfo.Email == "" {
+		return nil, &apperrors.ExternalAPIError{
+			Err:       errors.New("requester info fetch was not successful when checking for LCID expiration"),
+			Model:     intake,
+			ModelID:   intake.ID.String(),
+			Operation: apperrors.Fetch,
+			Source:    "CEDAR LDAP",
+		}
+	}
+
+	// TODO: Add other members of project team that need to be notified
+	emailsToNotify := []models.EmailAddress{requesterInfo.Email}
+
+	recipients := models.EmailNotificationRecipients{
+		RegularRecipientEmails:   emailsToNotify,
+		ShouldNotifyITGovernance: true,
+		ShouldNotifyITInvestment: false,
+	}
+
+	return &recipients, nil
+}
+
+// Iterates through all intakes to check if LCID is expiring with the next 60 days
+// If LCID is expiring with 60 days, an alert should be sent once and then again if LCID gets with 46 days (two weeks later) of expiration
+func checkForLCIDExpiration(
+	ctx context.Context,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+	fetchSystemIntakes func(context.Context) (models.SystemIntakes, error),
+	updateSystemIntake func(context.Context, *models.SystemIntake) (*models.SystemIntake, error),
+	sendLCIDExpirationEmail func(
+		ctx context.Context,
+		recipients models.EmailNotificationRecipients,
+		systemIntakeID uuid.UUID,
+		projectName string,
+		requesterName string,
+		lcid string,
+		lcidExpirationDate *time.Time,
+		scope string,
+		lifecycleCostBaseline string,
+		nextSteps string,
+	) error,
+) error {
```

Comment:
I wonder if we could write unit tests for this function by making `now` a `time.Time` parameter (rather than using `time.Now()` inside this function. This might help validate all the branching logic involved in this method.

Otherwise, I think these changes are good as-is, but testing this function would be helpful in validating that it behaves the way we want